File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefObjectProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.convert.Converter;
import gargoyle.ct.pref.CTPreferencesProvider;
import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

public class CTPrefObjectProperty<T extends Serializable> extends CTPrefProperty<T> {
    public CTPrefObjectProperty(Class<T> type, @NotNull CTPreferencesProvider provider, String name, @NotNull Converter<T> converter) {
        this(type, provider, name, null, converter);
    }

    public CTPrefObjectProperty(Class<T> type, @NotNull CTPreferencesProvider provider, String name, T def, @NotNull Converter<T> converter) {
        super(type, converter, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefObjectProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.convert.Converter
import gargoyle.ct.pref.CTPreferencesProvider
import kotlin.reflect.KClass

class CTPrefObjectProperty<T : Any>(
    type: KClass<T>,
    provider: CTPreferencesProvider,
    name: String,
    def: T?,
    converter: Converter<T>
) : CTPrefProperty<T>(type, converter, provider, name, def) {
    @Suppress("UNCHECKED_CAST")
    constructor(type: KClass<T>, provider: CTPreferencesProvider, name: String, converter: Converter<T>) : this(
        type,
        provider,
        name,
        null,
        converter
    )
}
